### ISSUE TITLE
Different `max_active_runs` for scheduled vs manual runs

### ISSUE THREAD

**Issue by hterik** [apache/airflow#44336]
### Description

Hi
We have a use case where a DAG is scheduled similar to `ContinuousTimetable`, where it should run as often as possible, but it shouldn't be scheduled to run more than one at a time. 
The same DAG can be started manually, with a set of parameters to override the default behaviour. When someone decides to start manually, it is OK to have 2 or more of them running concurrently.

I can't find any way to implement this, setting `max_active_runs=1`, constrains also the manual runs.

I would like a new parameter (`max_active_scheduled_runs`?), or some new extension point in the `TimeTable` API, that can take all ongoing runs into consideration and use that to withhold scheduling new runs.

### Use case/motivation

Some example use cases

* The scheduled runs are running a complete test suite that takes many hours. Using Params, a user can select one specific test to run, that will only take minutes to run.
* Shortening lead times. Imagine a run taking 1 hour. Half an hour in you realize you need to do a new run with some new data that became available. Instead of waiting for the current run to finish or aborting the current run, you can manually override to start a second run in parallel.  Always having a half-hour cadence active is not desirable due to cost.

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by jroachgolf84** [comment]
@cmarteepants, curious if this is something there may be appetite for? I'm curious if the same thing might be applicable for runs triggered via the API.

**Comment by DianaHov** [comment]
We hit this in production and I would like to implement it.

Our case: a few hundred dynamically generated S3 downloader DAGs. Each must have at most one active *scheduled* run (they sync overlapping date ranges, so concurrent scheduled runs duplicate work), while manual reruns need to start immediately rather than queue behind the active one. `max_active_runs=1` gives the first half and breaks the second; `max_active_runs=N` gives the second and breaks the first.

Proposed shape:

- new optional DAG param `max_active_scheduled_runs`, backed by a nullable `DagModel` column
- `max_active_runs` keeps its current meaning: the overall ceiling across all run types
- `max_active_scheduled_runs` additionally caps `run_type=scheduled` only
- default `None` → current behaviour exactly

I have this working and tested locally against 3.3.1 across our DAGs: scheduled runs never overlap, manual runs are unaffected.

One thing worth flagging: this needs to gate run *creation*, not just promotion. Gating only the promotion side leaves blocked scheduled runs accumulating in QUEUED, which isn't what people expect from a concurrency limit.

On @jroachgolf84's question about API-triggered runs: those are already covered by this shape. `trigger_dag_run` creates them with `run_type=MANUAL` (only `triggered_by` distinguishes UI from REST_API), so they wouldn't be constrained by `max_active_scheduled_runs`, same as UI triggers.

One design question before writing the PR: should asset-triggered runs count as scheduled? I assume not.

Note that backfills already have an independent `Backfill.max_active_runs` and the running counts are already grouped by `backfill_id`, so this extends an existing concept rather than introducing a new one.

Happy to put up the PR - just let me know on that point.

**Comment by jroachgolf84** [comment]
Yes, I think that Asset-triggered runs should be counted as scheduled. They go through the Scheduler loop I believe.

**Comment by DianaHov** [comment]
Thanks, that makes sense - I'll count asset-triggered runs as scheduled if this goes ahead.

**Comment by cmarteepants** [comment]
> or some new extension point in the `TimeTable` API

How viable would this path be?

**Comment by cmarteepants** [comment]
Also, thanks for raising the convo and questions here rather just opening a PR. We need more of that :) 

For context on where I'm coming from: this would be the third overlapping run-concurrency control on a DAG alongside `max_active_runs` and `Backfill.max_active_runs`, and params like that are hard to walk back once released. The asset-triggered question makes me think the semantics are broader than they look. Although I could go either way, my instinct was the opposite of Jake's, and I can see some people wanting to include it and some wanting to exclude it.

Which is why I'm curious about the timetable path. If it can extended to express what you need, it keeps this available to the people who want it and would allow you to tune it to your use-case without adding surface area for everyone else.

**Comment by DianaHov** [comment]
Thanks for picking this up so quickly, appreciate it.

The timetable route was actually where I started. What stopped me was that `next_dagrun_info()` has no visibility of current runs, and going to the DB from there disrupted the scheduler session - that's what pushed me towards a DAG-level param instead, rather than any preference for one over the other.

So if an extension point closes that gap, it would cover our case.

Either way this is something we really need on our side, so I'd like to see it through.

**Comment by cmarteepants** [comment]
Got it, makes sense. 

@uranusjr I think you're the closest to this: Can we get what @DianaHov needs for custom timetables into `next_dagrun_info()` without causing problems? Or add some other extension point?
